www-misc/buku/buku-4.2.2.ebuild

```diff
@@ -0,0 +1,96 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{5,6} )
+
+inherit bash-completion-r1 distutils-r1
+
+DESCRIPTION="Powerful command-line bookmark manager"
+HOMEPAGE="https://github.com/jarun/Buku"
+SRC_URI="test? ( https://github.com/jarun/Buku/files/1319933/bookmarks.zip -> buku-bookmarks.zip )"
+
+if [[ ${PV} == *9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/jarun/Buku"
+else
+	SRC_URI="${SRC_URI} mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz -> ${P}.tar.gz"
+
+	KEYWORDS="~amd64 ~x86"
+fi
+
+LICENSE="GPL-3"
+SLOT="0"
+IUSE="test server"
+RESTRICT="!test? ( test )"
+
+CLIENT_DEPEND="dev-python/beautifulsoup:4[${PYTHON_USEDEP}]
+	dev-python/certifi[${PYTHON_USEDEP}]
+	dev-python/cryptography[${PYTHON_USEDEP}]
+	dev-python/html5lib[${PYTHON_USEDEP}]
+	>=dev-python/urllib3-1.13.1[${PYTHON_USEDEP}]"
+
+SERVER_DEPEND="dev-python/arrow[${PYTHON_USEDEP}]
+	dev-python/click[${PYTHON_USEDEP}]
+	dev-python/flask[${PYTHON_USEDEP}]
+	dev-python/flask-admin[${PYTHON_USEDEP}]
+	dev-python/flask-api[${PYTHON_USEDEP}]
+	dev-python/flask-bootstrap[${PYTHON_USEDEP}]
+	dev-python/flask-paginate[${PYTHON_USEDEP}]
+	dev-python/flask-wtf[${PYTHON_USEDEP}]
+	dev-python/requests[${PYTHON_USEDEP}]
+	dev-python/werkzeug[${PYTHON_USEDEP}]"
+
+RDEPEND="${CLIENT_DEPEND}
+	server? ( ${SERVER_DEPEND} )"
+
+DEPEND="${RDEPEND}
+	test? (
+		dev-python/attrs[${PYTHON_USEDEP}]
+		dev-python/hypothesis[${PYTHON_USEDEP}]
+		dev-python/py[${PYTHON_USEDEP}]
+		dev-python/pytest[${PYTHON_USEDEP}]
+		dev-python/pyyaml[${PYTHON_USEDEP}]
+		dev-python/vcrpy[${PYTHON_USEDEP}]
+	)
+"
+
+python_prepare_all() {
+	distutils-r1_python_prepare_all
+
+	if ! use server; then
+		eapply "${FILESDIR}/buku-4.1-disable-server.patch" || die
```

Comment:
Conditional patching is a very bad idea. This patch is going to fail anytime upstream touches surrounding code, and if you build with USE=server you won't even notice. Also, `eapply` dies on its own.

My suggestion would be to prepare a proper configurable option for upstream (e.g. via setup.cfg), and submit it there.

Reply:
Did you mean something like this? I never used setup.cfg before. I will open a PR upstream with this patc

```
diff --git a/setup.cfg b/setup.cfg
new file mode 100644
index 0000000..9e16a70
--- /dev/null
+++ b/setup.cfg
@@ -0,0 +1,9 @@
+[options.extras_require]
+tests = tests_require_server_require
+server = server_require
+packaging = twine>=1.11.0
+
+[options.entry_points]
+console_scripts =
+   bukui = buku:main
+   bukuserver = bukuserver.server:cli
diff --git a/setup.py b/setup.py
index 7bd7cbd..e79154e 100644
--- a/setup.py
+++ b/setup.py
@@ -79,14 +79,6 @@ setup(
     ],
     packages=find_packages(exclude=['tests']),
     include_package_data=True,
-    entry_points={
-        'console_scripts': ['buku=buku:main', 'bukuserver=bukuserver.server:cli']
-    },
-    extras_require={
-        'tests': tests_require + server_require,
-        'server': server_require,
-        'packaging': ['twine>=1.11.0']
-    },
     test_suite='tests',
     tests_require=tests_require,
     keywords='cli bookmarks tag utility',
```

Reply:
I will try to fix this upstream for the next release. For now it's okay to have a patch?